bmm/src/main/java/org/openehr/bmm/v2/validation/converters/BmmModelCreator.java

```diff
@@ -12,72 +12,61 @@
         BmmModel model = new BmmModel();
-        model.setRmPublisher(schema.getRmPublisher());
-        model.setRmRelease(schema.getRmRelease());
-        model.setModelName(schema.getModelName());
-        model.setSchemaName(schema.getSchemaName());
-        model.setSchemaRevision(schema.getSchemaRevision());
-        model.setSchemaAuthor(schema.getSchemaAuthor());
-        model.setSchemaDescription(schema.getSchemaDescription());
-        model.setSchemaLifecycleState(schema.getSchemaLifecycleState());
-        //cannot set the documentation - the supported P_BMM version has no documentation in the P_BMM_SCHEMA
+        model.setRmPublisher (schema.getRmPublisher());
+        model.setRmRelease (schema.getRmRelease());
+        model.setModelName (schema.getModelName());
+        model.setSchemaName (schema.getSchemaName());
+        model.setSchemaRevision (schema.getSchemaRevision());
+        model.setSchemaAuthor (schema.getSchemaAuthor());
+        model.setSchemaDescription (schema.getSchemaDescription());
+        model.setSchemaLifecycleState (schema.getSchemaLifecycleState());
+        // cannot set the documentation - the supported P_BMM version has no documentation in the P_BMM_SCHEMA
         model.setSchemaContributors(schema.getSchemaContributors() == null ? new ArrayList() : new ArrayList<>(schema.getSchemaContributors()));
 
-
-
-        //Add packages first
+        // Add packages first
         for(PBmmPackage pBmmPackage:validationResult.getCanonicalPackages().values()) {
-            BmmPackage bmmPackage = createBmmPackageDefinition(pBmmPackage, null, null);
+            BmmPackage bmmPackage = createBmmPackageDefinition (pBmmPackage, null, null);
 
             model.addPackage(bmmPackage);
 
             pBmmPackage.doRecursiveClasses((p, s) -> {
-                PBmmClass persistedBmmClass = schema.findClassOrPrimitiveDefinition(s);
-                if (persistedBmmClass != null) {
-                    BmmClass bmmClass = classCreator.createBmmClass(persistedBmmClass);
-
-                    if (bmmClass != null && bmmPackage != null) {
-                        if (schema.getPrimitiveTypes().get(bmmClass.getName()) != null) {
-                            bmmClass.setPrimitiveType(true);
-                        }
-                        if (persistedBmmClass.isOverride() != null && persistedBmmClass.isOverride()) {
-                            bmmClass.setOverride(true);
-                        }
+                PBmmClass pBmmClass = schema.getClassDefinition(s);
+                if (pBmmClass != null) {
+                    BmmClass bmmClass = pBmmClass.createBmmClass();
+                    if (bmmClass != null) {
+                        bmmClass.setPrimitiveType(schema.getPrimitiveTypes().containsKey (bmmClass.getName()));
+                        bmmClass.setOverride(pBmmClass.isOverride() == null ? false : pBmmClass.isOverride());
```

Comment:
this setOverride() should probably be moved to createBmmClass(), set PrimitiveTypes is fine as is.